src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RangeAttribute.cs

```diff
@@ -94,10 +107,15 @@ public RangeAttribute(double minimum, double maximum)
 
         private void Initialize(IComparable minimum, IComparable maximum, Func<object, object?> conversion)
         {
-            if (minimum.CompareTo(maximum) > 0)
+            int cmp = minimum.CompareTo(maximum);
+            if (cmp > 0)
             {
                 throw new InvalidOperationException(SR.Format(SR.RangeAttribute_MinGreaterThanMax, maximum, minimum));
             }
+            else if (cmp == 0 && (MinimumIsExclusive || MaximumIsExclusive))
```

Comment:
What about the case of e.g. minimum==1, maximum==2 and both are exclusive?

Reply:
I considered it but ultimately didn't want to introduce validation that is specific to integers. Other types might also define discrete orders but we wouldn't be able to validate these.

Reply:
So we're just trying to flag some mistakes a dev might make and we're ok with false negatives, is that right?

Reply:
Correct, it would be impossible to detect every empty range for arbitrary comparers. I could try special casing a check for integers but it seems unlikely people would use exclusive bounds on integer types.